server/error_code/error_code.go

```diff
@@ -0,0 +1,72 @@
+// Package errcode is designed to create standardized API error codes.
+// The goal is that clients can reliably check against immutable error codes
+//
+// Two approaches can be takend
+// 1) centralized: define all errors in one module
+// 2) modular: define errors where they occur
+//
+// The centralized approach helps organize information.
+// All errors can be found in one file, and changes can be carefully reviewed.
+// The downside of centralized is the potential need to import types from other packages.
+//
+// A RegisteredCode should never be modified once committed (and released)
+package errcode
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// RegisteredCode helps document that we are using a registered error code that must never change
+type RegisteredCode string
+
+const (
+	// InternalErrorCode means the operation placed the system is in an inconsistent or unrecoverable state
+	// Essentially a handled panic.
+	// This is the same as a HTTP 500, so it is not necessary to send this code when using HTTP.k
+	InternalErrorCode RegisteredCode = "internal"
```

Comment:
`InternalErrorCode` defined but not use it in this PR, will do it in next PR?

Reply:
Yes, the idea is we can transition to using error codes at the pace we like, although it would probably be good to demonstrate using this once. Unrelated. it looks like I have a stray character in the comment here.